--- original/api-functions-insertsalesline.sql	2012-05-11 13:39:01.000000000 +0800
+++ api-functions-insertsalesline.sql	2012-05-11 13:48:22.000000000 +0800
@@ -1,6 +1,10 @@
-CREATE OR REPLACE FUNCTION insertSalesLine(api.salesline) RETURNS boolean AS $$
--- Copyright (c) 1999-2012 by OpenMFG LLC, d/b/a xTuple. 
--- See www.xtuple.com/CPAL for the full text of the software license.
+-- Function: insertsalesline(api.salesline)
+
+-- DROP FUNCTION insertsalesline(api.salesline);
+
+CREATE OR REPLACE FUNCTION insertsalesline(api.salesline)
+  RETURNS boolean AS
+$BODY$
 DECLARE
   pNEW ALIAS FOR $1;
   _r RECORD;
@@ -67,7 +71,7 @@
     COALESCE(getUomId(pNEW.qty_uom),_r.item_inv_uom_id),
     itemuomtouomratio(_r.item_id,COALESCE(getUomId(pNEW.qty_uom),_r.item_inv_uom_id),_r.item_inv_uom_id),
     0,
-    stdCost(_r.item_id),
+    itemcost_dispense(_r.item_id, _r.qty_ordered),
     COALESCE(pNEW.net_unit_price,itemPrice(_r.item_id,_r.cohead_cust_id,
              _r.cohead_shipto_id,pNEW.qty_ordered,_r.cohead_curr_id,_r.cohead_orderdate)),
     COALESCE(getUomId(pNEW.price_uom),_r.item_price_uom_id),
@@ -98,4 +102,7 @@
 
   RETURN TRUE;
 END;
-$$ LANGUAGE 'plpgsql';
+$BODY$
+  LANGUAGE plpgsql VOLATILE
+  COST 100;
+ALTER FUNCTION insertsalesline(api.salesline) OWNER TO "admin";
