--- original/distributetolocations.sql	2012-05-11 13:23:56.000000000 +0800
+++ distributetolocations.sql	2012-05-09 08:48:38.000000000 +0800
@@ -1,6 +1,10 @@
-CREATE OR REPLACE FUNCTION distributeToLocations(INTEGER) RETURNS INTEGER AS $$
--- Copyright (c) 1999-2012 by OpenMFG LLC, d/b/a xTuple. 
--- See www.xtuple.com/CPAL for the full text of the software license.
+-- Function: distributetolocations(integer)
+
+-- DROP FUNCTION distributetolocations(integer);
+
+CREATE OR REPLACE FUNCTION distributetolocations(integer)
+  RETURNS integer AS
+$BODY$
 DECLARE
   pItemlocdistid ALIAS FOR $1;
   _distCounter INTEGER;
@@ -167,7 +171,7 @@
              uom_name, stdCost(item_id),
              itemsite_costmethod, itemsite_value,
              (itemsite_value + (_itemlocdist.qty * -1 * CASE WHEN(itemsite_costmethod='A') THEN avgcost(itemsite_id)
-                                                             ELSE stdCost(itemsite_item_id)
+                                                             ELSE itemcost_dispense(itemsite_item_id, _itemlocdist.qty)
                                                         END)),
              _itemlocdist.series
       FROM item, itemsite, invhist, uom
@@ -224,4 +228,7 @@
   RETURN _distCounter;
 
 END;
-$$ LANGUAGE 'plpgsql';
+$BODY$
+  LANGUAGE plpgsql VOLATILE
+  COST 100;
+ALTER FUNCTION distributetolocations(integer) OWNER TO "admin";
